### PR TITLE
Moved the email notification logic to a seperate schedule

### PR DESCRIPTION
**Pull request by Hrishabh17**
Please review.

### CODE REVIEW

File: apps/bamboohr/views.py [hunk 1]
```diff
@@ -196,6 +196,8 @@ def post(self, request, *args, **kwargs):
     
         async_task('apps.bamboohr.tasks.import_employees', kwargs['org_id'])
 
+        schedule_failure_emails_for_employees(kwargs['org_id'])
```
Comment (ashwin1111) [inline apps/bamboohr/views.py:199]: this has to be created where we create other schedule where we import employees

File: fyle_employee_imports/base.py [hunk 2]
```diff
@@ -169,3 +170,42 @@ def sync_employees(self, is_incremental_sync: bool):
 
         self.import_departments(hrms_employees)
         self.fyle_employee_import(hrms_employees)
+
+    def send_employee_email_missing_failure_notification(self):
```
Comment (ashwin1111) [inline fyle_employee_imports/base.py:174]: we don't have to run all functions when this job runs, we just need to check all destination where email is null and is_failure_email_sent is false and then send emails for those emails and mark these rows as is_failure_email_sent -> true

File: apps/fyle_hrms_mappings/models.py [hunk 3]
```diff
@@ -108,22 +111,27 @@ def bulk_create_or_update_destination_attributes(
                         ('detail' in attribute and attribute['detail'] != primary_key_map[attribute['destination_id']]['detail'])
                         or
                         ('active' in attribute and attribute['active'] != primary_key_map[attribute['destination_id']]['active'])
+                        or
+                        (
+                            'is_failure_email_sent' in attribute and attribute['is_failure_email_sent'] != primary_key_map[attribute['destination_id']]['is_failure_email_sent']
```
Comment (ashwin1111) [inline apps/fyle_hrms_mappings/models.py:116]: if the below comments are taken care of, then we don't have to specify these things

File: apps/fyle_hrms_mappings/models.py [hunk 4]
```diff
@@ -115,15 +116,15 @@ def bulk_create_or_update_destination_attributes(
                                 value=attribute['value'],
                                 detail=attribute['detail'] if 'detail' in attribute else None,
                                 active=attribute['active'] if 'active' in attribute else None,
-                                updated_at=datetime.now()
+                                updated_at=datetime.now(),
```
Comment (NileshPant1999) [inline apps/fyle_hrms_mappings/models.py:119]: why is the `,` added ??

File: apps/fyle_hrms_mappings/models.py [hunk 5]
```diff
@@ -115,15 +116,15 @@ def bulk_create_or_update_destination_attributes(
...
                             )
                         )
         if attributes_to_be_created:
             DestinationAttribute.objects.bulk_create(attributes_to_be_created, batch_size=50)
 
         if attributes_to_be_updated:
             DestinationAttribute.objects.bulk_update(
-                attributes_to_be_updated, fields=['detail', 'value', 'active', 'updated_at'], batch_size=50)
+                attributes_to_be_updated, fields=['detail', 'value', 'active', 'updated_at',], batch_size=50)
```
Comment (NileshPant1999) [inline apps/fyle_hrms_mappings/models.py:127]: here too

File: apps/fyle_hrms_mappings/models.py [hunk 6]
```diff
@@ -21,6 +21,7 @@ class DestinationAttribute(models.Model):
     destination_id = models.CharField(max_length=255, help_text='Destination ID')
     auto_created = models.BooleanField(default=False,
                                         help_text='Indicates whether the field is auto created by the integration')
+    is_failure_email_sent = models.BooleanField(default=False, help_text='Indicates whether the failure email is sent')
```
Comment (ashwin1111) [inline apps/fyle_hrms_mappings/models.py:24]: write script to fill existing failed rows as true

File: fyle_employee_imports/base.py [hunk 7]
```diff
@@ -169,3 +170,25 @@ def sync_employees(self, is_incremental_sync: bool):
...
+        hrms_employees = DestinationAttribute.objects.filter(
+            Q(detail__email=None) | Q(detail__email=''),
```
Comment (ashwin1111) [inline fyle_employee_imports/base.py:176]: detail__email__isnull=True

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**